image/manifest.go

```diff
@@ -89,7 +89,7 @@ func (m *manifest) validate(w walker) error {
 
 func (m *manifest) unpack(w walker, dest string) error {
 	for _, d := range m.Layers {
-		if d.MediaType != string(schema.MediaTypeImageConfig) {
+		if d.MediaType != string(schema.MediaTypeImageLayer) {
 			continue
```

Comment:
On Wed, Sep 07, 2016 at 08:46:36PM -0700, Lei Jitang wrote:
> > @@ -89,7 +89,7 @@ func (m *manifest) validate(w walker) error {
> >  
> >  func (m *manifest) unpack(w walker, dest string) error {
> >  	for _, d := range m.Layers {
> > -		if d.MediaType != string(schema.MediaTypeImageConfig) {
> > +		if d.MediaType != string(schema.MediaTypeImageLayer) {
> >  			continue
> 
> @wking If the spec does not limit the allowed types, I don't think
> we should error out on unpacking, we should just unpacking the
> allowed typed layers.  But I think this is another issue we should
> discuss more not related to this pr

If there's any confusion, I'm fine punting this to another PR.  The
media type fix is critical and obviously right.  The continue→error
fix only matters for oddball manifests, so it's much less important.